### ISSUE TITLE
Boot from image, specifically ISO

### ISSUE THREAD

**Issue by jgeorgeson** [gophercloud/gophercloud#840]
I'd like to see a Packer openstack-iso builder (ie boot an instance with an ISO image and run kickstart file to build stemcell OS images) or for the existing openstack builder to be enabled with such functionality. 

https://github.com/hashicorp/packer/issues/3705

Seems the openstack builder uses this library, so I think the first step would be for gophercloud to have a `bootfromimage` extension. While the request and use-case are specifically for boot from ISO, the CLI docs show that it's just a specialized case of booting from an image

https://docs.openstack.org/ocata/user-guide/cli-nova-launch-instance-using-ISO-image.html
https://docs.openstack.org/ocata/user-guide/cli-nova-launch-instance-from-image.html

**Comment by jtopjian** [comment]
@jgeorgeson I'm surprised this doesn't work in Gophercloud. Are you sure this is an issue specifically to Gophercloud or is it just in the way that Packer is using it?

One way for us to verify if support is in Gophercloud is if you could provide the output of:

```shell
$ openstack --debug server create ...
```

And then provide the (sanitized) output to a github gist.

**Comment by jgeorgeson** [comment]
I guess I forget to mention I was speculating based on the presence of a `bootfromvolume` extension and the lack of a `bootfromimage` extension that it's not currently supported by gophercloud. I'm not really a Go developer and I'm pretty new to OpenStack. Looking at the  CLI for booting from a volume

https://docs.openstack.org/ocata/user-guide/cli-nova-launch-instance-from-volume.html

I see that is also a specialized case of booting from an image. And _booting_ seems to be a misnomer as in all cases it's creating a new instance from an image. So calling the `Create()` method from compute/v2/servers/requests with the ISO as the source image listed in the create options should be equivalent to specifying an ISO with `openstack server create --image foo.iso ...`. Is that correct?

**Comment by jtopjian** [comment]
@jgeorgeson Yeah, it's a little confusing. Booting from an image is the default behavior, which is why there's no explicit extension. Booting from a volume is indeed a specialized case, hence the extension.

> So calling the Create() method from compute/v2/servers/requests with the ISO as the source image listed in the create options should be equivalent to specifying an ISO with openstack server create --image foo.iso .... Is that correct?

Yep. In theory. :)

I've personally never booted an ISO so there's always a slight chance that the `openstack` tool is doing some customization behind the scenes when it detects that an ISO was requested. 

If you aren't in a position to test that out, I can certainly try at some point. It might take me a few days to get to.
